openpype/hosts/maya/plugins/publish/validate_arnold_scene_source_cbid.py

```diff
@@ -0,0 +1,73 @@
+import pyblish.api
+from openpype.hosts.maya.api import lib
+from openpype.pipeline.publish import (
+    ValidateContentsOrder, PublishValidationError, RepairAction
+)
+
+
+class ValidateArnoldSceneSourceCbid(pyblish.api.InstancePlugin):
+    """Validate Arnold Scene Source Cbid.
+
+    It is required for the proxy and content nodes to share the same cbid.
+    """
+
+    order = ValidateContentsOrder
+    hosts = ["maya"]
+    families = ["ass"]
+    label = "Validate Arnold Scene Source CBID"
+    actions = [RepairAction]
+
+    @staticmethod
+    def _get_nodes_data(nodes):
+        nodes_by_name = {}
+        for node in nodes:
+            node_name = node.rsplit("|", 1)[-1].rsplit(":", 1)[-1]
+            nodes_by_name[node_name] = node
+
+        return nodes_by_name
+
+    def get_invalid_couples(self, instance):
+        content_nodes_by_name = self._get_nodes_data(
+            instance.data["contentMembers"]
+        )
+        proxy_nodes_by_name = self._get_nodes_data(
+            instance.data.get("proxy", [])
+        )
+
+        invalid_couples = []
+        for content_name, content_node in content_nodes_by_name.items():
+            proxy_node = proxy_nodes_by_name.get(content_name, None)
+
```

Comment:
This logic would still allow proxy nodes to exist which do not exist in content.

We might want to do:
```python
names = set(content_nodes_by_name) + set(proxy_nodes_by_name)
for name in names:
    proxy_node = proxy_nodes_by_name.get(name, None)
    content_node = content_nodes_by_name.get(name, None)
```

And then check for both `content_node` and `proxy_node` whether they exist.

Reply:
We validate we have the same amount of content to proxy nodes [here](https://github.com/tokejepsen/pype/blob/feature/OP-4859_cant-assign-shaders-to-the-ass-file/openpype/hosts/maya/plugins/publish/validate_arnold_scene_source.py#L84-L92)